### PR TITLE
Major overhaul of update checking

### LINKED ISSUE #333: a few design issues/optimizations
- [x] left menue: Donate link should open in a new window/tab
- [x] footer: Donate link should open in a new window/tab
- [x] footer: on small devices/browser windows: information " Pi-hole Version v2.11 Web Interface Version v2.3 " is missing
- [x] layer on click pi-hole button: Links "GitHub, Details, Updates" should open in a new window/tab
- [x] layer on click pi-hole button: remove square around logo - not consistent with logo in header, theres no square either
- [x] layer on click pi-hole button: make 23:48 (time information) bold in String "Session is valid for 23:48"
- [x] layer on click pi-hole button: on small devices/browser windows not alignd right (see screenshot1.png)
- [x] on small devices/browser windows: design issue left side (see screenshot1.png)
- [x] on some smaller devices/browser windows: the text is behind the "Log in" Button" (see screenshot2.png and screenshot3.png)
- [x] left menu: menu/site indicator stays on "main page" althoug i am on the login page (see screenshot4.png)

screenshot1
screenshot2
screenshot3
screenshot4

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X] - no spaces) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

## 10

---

Major overhaul of update checking on the dash board

Before this PR:
- Each browser refresh initiates a request to GitHub's API and queries the current version to compare it to the local version

With this PR:
- On the first page loading after you enabled this PR, the current versions are queried from GitHub and stored (along a time stamp of the request) in the file `/var/www/html/versions` (this is the only folder where `lighttpd` has unlimited write permissions by default). For the next 30 minutes, the upstream versions are queried from this file *instead* of reaching out to GitHub. If the version strings in the file are older than 30min, then PHP will query the data from GitHub's API and update the values in the `versions` file.
- If the file cannot be created/does not exist/is not writable, we fall back to the current behavior (query versions on every page load).

Note: This removes the update display dependence on Javascript *altogether*. Everything is now done server-side from PHP.

It also partially fixes #333 as it shows the versions/branches/update notifications in the dropdown menu.

Screenshot on PC (Vivaldi + Linux):
da5f1247-681c-4807-9864-e85038720104

Screenshot an Android (Chrome Mobile):
pi hole-admin-index php-login galaxy s5 2

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: scripts/pi-hole/php/header.php [hunk 1]
```diff
@@ -264,11 +265,18 @@ function pidofFTL()
...
-                                    <a class="alert-link" href="https://github.com/pi-hole/AdminLTE/releases">There's an update available for this Web Interface!</a>
+                                <div>
+                                    <b>Pi-hole Version </b> <?php
+                                    echo $core_current;
+                                    if(isset($core_commit)) { echo "<br>(".$core_branch.", ".$core_commit.")"; }
+                                    if($core_update){ ?> <a class="alert-link lookatme" href="https://github.com/                        pi-hole/pi-hole/releases">(Update available!)</a><?php } ?><br>
+                                    <b>Web Interface Version </b><?php
+                                    echo $web_current;
+                                    if(isset($web_commit)) { echo "<br>(".$web_branch.", ".$web_commit.")"; }
+                                    if($web_update){ ?> <a class="alert-link lookatme\" href="https://github.com/                        pi-hole/AdminLTE/releases">(Update available!)</a><?php } ?><br>
```
Comment (WaLLy3K) [inline scripts/pi-hole/php/header.php]: Typo?

**Comment by DL6ER** [comment]
Another screenshot, @Th3M3
pi hole-admin-index php-login nexus 6p

File: scripts/pi-hole/php/footer.php [hunk 2]
```diff
@@ -77,10 +53,18 @@
...
-            <b>FTL Version </b> <span id="FTLVersion"><?php echo $FTLVersion; ?></span>
+        <div class="pull-right hidden-xs <?php if(isset($core_commit) || isset($web_commit)) { ?>hidden-md<?php } ?>">
+            <b>Pi-hole Version </b> <?php
+            echo $core_current;
+            if(isset($core_commit)) { echo " (".$core_branch.", ".$core_commit.")"; }
+            if($core_update){ ?> <a class="alert-link lookatme" href="https://github.com/pi-hole/pi-hole/releases">(Update available!)</a><?php } ?>
+            <b>Web Interface Version </b><?php
+            echo $web_current;
+            if(isset($web_commit)) { echo " (".$web_branch.", ".$web_commit.")"; }
+            if($web_update){ ?> <a class="alert-link lookatme\" href="https://github.com/pi-hole/AdminLTE/releases">(Update available!)</a><?php } ?>
```
Comment (WaLLy3K) [inline scripts/pi-hole/php/footer.php]: Is `lookatme\` a typo?

**Comment by Th3M3** [comment]
@DL6ER The dropdown-menu looks similar to my screenshot :-)
Actually I have updated to the current devel branch.

My suggestion was to show the version info in the dropdown-menu _or_ in the footer, depending to the display width.

**Comment by DL6ER** [comment]
> My suggestion was to show the version info in the dropdown-menu or in the footer, depending to the display width.

I know but that didn't work reliably on the stock browser of my Android device. Hence, I decided to have it there always (I barely ever open this menu).

**Comment by Th3M3** [comment]
> I know but that didn't work reliably on the stock browser of my Android device.

Hmm, ok.
For me it was working when adding classes `hidden-md` and `hidden-lg` to the `div` element

**Comment by SchneiderDe** [comment]
Hello DL6ER,

At the moment, I have deactivated the DNS of the flowing outbound connections and mapped it with the IP:127.0.0.1 with three such vigilance applications installed on my workstation for such a purpose. Thereafter, any outbound connections generated dies immediately upon its birth, regardless of if it is configured 30 or 60 minutes.

With the current - and the future version after this PR -, one still requires to block the flowing river of outbound connections from a user's workstation, if one wants to prevent registration of one's online activities "somewhere" by ones, who may want to trap it. 

"With this PR: On the first page loading after you enabled this PR, the current versions are queried from GitHub and stored ..."

It would be helpful to others to **offer an option to deactivate an automatic checking and turn on such a check manually**. By doing this, all pi-Hole users are not compelled to swim in the river of outbound connections and could easily construct a dam on this river!

**Comment by dschaper** [comment]
SchneiderDe, this PR has been merged so no additional pulls or commits to the branch would be feasible. I would suggest that any feature requests be handled via Discourse where we track those issues. Comments here will be lost as we don't actively follow or comment on merged PR's. 

Thank you for your understanding.

**Comment by SchneiderDe** [comment]
@dschaper,

> I would suggest that any feature requests be handled via Discourse where we track those issues.

Thanks for drawing my attention to it. I have missed it.

And how do you expect me to discuss my opinion via Discourse, when you have put my account - for no logical reasons - on hold there? Or did the system put it on hold automatically for some reasons?
There is - or was - nothing offensive to anyone or anywhere in my messages.
So, I was surprised and fail to understand why my account has been put on hold!

**Comment by dschaper** [comment]
Please email me privately at dan.schaper@pi-hole.net to discuss any issues you may have. PR conversations are not places to air grievances.

**Pull request merged**